lib/azure/resource_management/ARM_interface.rb

```diff
@@ -148,6 +148,15 @@ def show_server(name, resource_group)
 
           unless result.nil?
             server = result.body
+            network_interface_name = server.properties.network_profile.network_interfaces[0].id.split('/')[-1]
+            network_interface_data = network_resource_client.network_interfaces.get(resource_group, network_interface_name).value!.body
+            public_ip_id_data = network_interface_data.properties.ip_configurations[0].properties.public_ipaddress
+            unless public_ip_id_data.nil?
+              public_ip_name = public_ip_id_data.id.split('/')[-1]
+              public_ip_data = network_resource_client.public_ipaddresses.get(resource_group, public_ip_name).value!.body
+            else
+              public_ip_data = nil

```

Comment:
`public_ip_data` variable is not  defined before hand, it gets defined in `unless` or `else` part and gets referenced in the code , so if we remove the else part, the code will fail in the scenario where  `public_ip_id_data` is nil.